pages/api/student_email_join.js

```diff
@@ -1,37 +1,59 @@
 import prisma from '../../prisma/prisma';
+import { unstable_getServerSession } from 'next-auth';
+import { authOptions } from './auth/[...nextauth]';
 
 export default async function handle(req, res) {
+  // unstable_getServerSession is recommended here: https://next-auth.js.org/configuration/nextjs
+  const session = await unstable_getServerSession(req, res, authOptions);
+
+  if (!req.method == 'PUT') {
+    res.status(405).end();
+  }
+
+  if (!session) {
+    res.status(403).end();
+  }
+
   const data = JSON.parse(req.body);
```

Comment:
I think we decided earlier that form body should be JSON and we shouldn't need to parse JSON from plaintext going forward right?

Reply:
fixed.